### ISSUE TITLE
Rule for naming functions inside function

### ISSUE THREAD

**Issue by Greesb** [astral-sh/ruff#9786]
Hello,

Just thought about a potential rule that would be triggered when a function inside a function doesn't have a specific (configurable) prefix, eg:
```python
def my_outer_func():
    # maybe some code maybe not
    def idk():  # error: function inside function should have the `inner_` prefix
        pass
    # maybe some code again maybe not
    idk()
    # maybe some code again again maybe not
```

What do you think ?
